File: 29Items-Projects/11DevOps/1-Java-Microservices-CICD/services/order-service/src/main/java/com/example/orderservice/exception/OrderNotFoundException.java
package com.example.orderservice.exception;

import java.util.UUID;

/** Raised when an order id does not exist; mapped to HTTP 404 by the web layer. */
public class OrderNotFoundException extends RuntimeException {

    private final UUID orderId;

    public OrderNotFoundException(UUID orderId) {
        super("Order %s was not found".formatted(orderId));
        this.orderId = orderId;
    }

    public UUID getOrderId() {
        return orderId;
    }
}

File: 29Items-Projects/15Php/2E-CommerceMarketplace/backend/src/Ordering/Domain/Model/OrderNotFoundException.php
<?php

declare(strict_types=1);

namespace App\Ordering\Domain\Model;

use App\Shared\Domain\Exception\NotFoundException;

/** Raised when an order stream does not exist in the event store. Mapped to 404. */
final class OrderNotFoundException extends NotFoundException
{
    public static function withId(OrderId $id): self
    {
        return new self(\sprintf('Order "%s" was not found.', $id));
    }
}
